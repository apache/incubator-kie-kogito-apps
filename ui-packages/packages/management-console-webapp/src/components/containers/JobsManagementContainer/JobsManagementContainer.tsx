/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { OUIAProps } from '@kogito-apps/components-common';
import { EmbeddedJobsManagement } from '@kogito-apps/jobs-management';
import { JobsManagementGatewayApi } from '../../../channel/JobsManagement';
import { useJobsManagementGatewayApi } from '../../../channel/JobsManagement/JobsManagementContext';

const JobsManagementContainer: React.FC<OUIAProps> = () => {
  const gatewayApi: JobsManagementGatewayApi = useJobsManagementGatewayApi();
  return (
    <EmbeddedJobsManagement
      driver={gatewayApi}
      targetOrigin={window.location.origin}
    />
  );
};

export default JobsManagementContainer;
