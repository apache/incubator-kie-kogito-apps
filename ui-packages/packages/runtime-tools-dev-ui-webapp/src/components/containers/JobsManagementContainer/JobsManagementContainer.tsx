import React from 'react';
import {
  componentOuiaProps,
  OUIAProps
} from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';
import { EmbeddedJobsManagement } from '@kogito-apps/jobs-management';
import { JobsManagementGatewayApi } from '../../../channel/JobsManagement';
import { useJobsManagementGatewayApi } from '../../../channel/JobsManagement/JobsManagementContext';
import { useDevUIAppContext } from '../../contexts/DevUIAppContext';

const JobsManagementContainer: React.FC<OUIAProps> = ({ ouiaId, ouiaSafe }) => {
  const gatewayApi: JobsManagementGatewayApi = useJobsManagementGatewayApi();
  const appContext = useDevUIAppContext();
  return (
    <EmbeddedJobsManagement
      driver={gatewayApi}
      targetOrigin={appContext.getDevUIUrl()}
      {...componentOuiaProps(ouiaId, 'jobs-management-container', ouiaSafe)}
    />
  );
};

export default JobsManagementContainer;
