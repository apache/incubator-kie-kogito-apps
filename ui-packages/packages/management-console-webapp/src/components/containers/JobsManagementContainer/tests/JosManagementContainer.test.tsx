import { mount } from 'enzyme';
import React from 'react';
import JobsManagementContainer from '../JobsManagementContainer';

describe('WebApp - JobsManagementContainer tests', () => {
  it('Snapshot test with default values', () => {
    const wrapper = mount(<JobsManagementContainer />).find(
      'JobsManagementContainer'
    );
    expect(wrapper).toMatchSnapshot();
  });
});
