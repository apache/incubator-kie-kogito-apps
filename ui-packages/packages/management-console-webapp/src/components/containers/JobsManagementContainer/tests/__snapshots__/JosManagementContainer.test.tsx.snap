// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebApp - JobsManagementContainer tests Snapshot test with default values 1`] = `
<JobsManagementContainer>
  <ForwardRef
    driver={null}
    targetOrigin="http://localhost"
  >
    <ForwardRef>
      <div />
    </ForwardRef>
  </ForwardRef>
</JobsManagementContainer>
`;
